niddler-lib-base/src/main/java/com/icapps/niddler/interceptor/okhttp/NiddlerOkHttpTimeoutResponse.java

```diff
@@ -0,0 +1,98 @@
+package com.icapps.niddler.interceptor.okhttp;
+
+import com.icapps.niddler.core.NiddlerRequest;
+import com.icapps.niddler.core.NiddlerResponse;
+
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * @author Nicola Verbeeck
+ */
+final class NiddlerOkHttpTimeoutResponse implements NiddlerResponse {
+
+	private final String mRequestId;
+	private final String mMessageId;
+	private final long mTimestamp;
+
+	NiddlerOkHttpTimeoutResponse(final String requestId) {
+		mRequestId = requestId;
+		mMessageId = UUID.randomUUID().toString();
+		mTimestamp = System.currentTimeMillis();
+	}
+
+	@Override
+	public String getMessageId() {
+		return mMessageId;
+	}
+
+	@Override
+	public String getRequestId() {
+		return mRequestId;
+	}
+
+	@Override
+	public long getTimestamp() {
+		return mTimestamp;
+	}
+
+	@Override
+	public Map<String, List<String>> getHeaders() {
+		return Collections.emptyMap();
+	}
+
+	@Override
+	public Integer getStatusCode() {
+		return 0;
```

Comment:
I think perhaps -1 or some other constant is a better option? 0 feels like it could be a valid code.

Reply:
Every status code can have some meaning for some protocol. This status code is not really used for anything except when exporting to charles